pyflakes/checker.py

```diff
@@ -425,7 +425,9 @@ def unusedAssignments(self):
         Return a generator for the assignments which have not been used.
         """
         for name, binding in self.items():
-            if (not binding.used and name not in self.globals
+            if (not binding.used
+                    and (name[0:1] != '_' or name[0:2] == '__')
```

Comment:
I'm uncertain about the (`name[0:1] != '_' or name[0:2] == '__'`) logic here. One is using `!=` while the other is using a `==`. Can you add additional test case to demonstrate what this is trying to do?

Reply:
Sorry added a comment elsewhere. There is another test case which fails without the second half of the or clause. It explicitly tests that __all__ defined in a local scope and not used should be an error. The bit after the 'or' was simply to get that test to pass again.

Reply:
I don't think this should be done as noted in another comment, but the clean way to write this is to use the `name.startswith` method.